syntax/scan.go

```diff
@@ -864,26 +863,24 @@ func (sc *scanner) scanNumber(val *tokenValue, c rune) Token {
 		return FLOAT
 	} else {
 		var err error
-		var ok bool = true
 		s := val.raw
+		val.bigInt = nil
 		if len(s) > 2 && s[0] == '0' && (s[1] == 'o' || s[1] == 'O') {
 			val.int, err = strconv.ParseInt(s[2:], 8, 64)
 		} else if len(s) > 2 && s[0] == '0' && (s[1] == 'b' || s[1] == 'B') {
 			val.int, err = strconv.ParseInt(s[2:], 2, 64)
 		} else {
-			// Checking int
 			val.int, err = strconv.ParseInt(s, 0, 64)
-			// Checking BigInt
 			if err != nil {
 				num := new(big.Int)
+				var ok bool = true
```

Comment:
var ok bool

There's no need to initialize it since the next statement will assign it.